src/components/mobile-nav-item/mobile-nav-item.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from 'react'
+import type { FC } from 'react'
+
+import { useId } from '../../storybook/random-id'
+
+import { Icon } from '../icon'
+
+import { MobileNavItemExpandableProps, MobileNavItemSimpleProps, MobileNavSubItemProps } from './types'
+import {
+  ElMobileNavItemAnchor,
+  ElMobileNavItemBadge,
+  ElMobileNavItemContent,
+  ElMobileNavItemExpanderButton,
+  ElMobileNavItemListItem,
+  ElMobileNavItemUnorderedList,
+  ElMobileNavSubItemUnorderedList,
+} from './styles'
+import { MobileNavSubItem } from './mobile-nav-sub-item'
+
+export type MobileNavItemProps = MobileNavItemSimpleProps | MobileNavItemExpandableProps
+
+const isExpandableVariant = (props: MobileNavItemProps): props is MobileNavItemExpandableProps => {
+  return 'children' in props
+}
+
+type MobileNavItemFC = FC<MobileNavItemProps> & {
+  SubItem: FC<MobileNavSubItemProps>
+}
+
+const MobileNavItem: MobileNavItemFC = (props) => {
+  const [isExpanded, setIsExpanded] = useState(Boolean(props?.isDefaultExpanded))
+  const panelId = useId()
+
+  if (isExpandableVariant(props)) {
```

Comment:
**suggestion:** I think there's enough going on here to warrant splitting out each branch of this if-else statement to separate subcomponents.